host/schedule_test.go

```diff
@@ -0,0 +1,541 @@
+			atomic.AddInt32(&runs, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflowFn, workflow.RegisterOptions{Name: wt})
+	workflow2Fn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs2, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflow2Fn, workflow.RegisterOptions{Name: wt2})
+
+	// create
+
+	createTime := time.Now()
+	_, err := s.engine.CreateSchedule(NewContext(), req)
+	s.NoError(err)
+
+	// sleep until we see two runs, plus a bit more
+	s.Eventually(func() bool { return atomic.LoadInt32(&runs) == 2 }, 8*time.Second, 200*time.Millisecond)
+	time.Sleep(100 * time.Millisecond)
+
+	// describe
+
+	describeResp, err := s.engine.DescribeSchedule(NewContext(), &workflowservice.DescribeScheduleRequest{
+		Namespace:  s.namespace,
+		ScheduleId: sid,
+	})
+	s.NoError(err)
+
+	s.Equal(schedule.Spec, describeResp.Schedule.Spec)
+	s.Equal(enumspb.SCHEDULE_OVERLAP_POLICY_SKIP, describeResp.Schedule.Policies.OverlapPolicy) // set to default value
+
+	s.Equal(schSAValue.Data, describeResp.SearchAttributes.IndexedFields[csa].Data)
+	s.Equal(schMemo.Data, describeResp.Memo.Fields["schedmemo1"].Data)
+	s.Equal(wfSAValue.Data, describeResp.Schedule.Action.GetStartWorkflow().SearchAttributes.IndexedFields[csa].Data)
+	s.Equal(wfMemo.Data, describeResp.Schedule.Action.GetStartWorkflow().Memo.Fields["wfmemo1"].Data)
+
+	s.DurationNear(describeResp.Info.CreateTime.Sub(createTime), 0, 500*time.Millisecond)
+	s.EqualValues(2, describeResp.Info.ActionCount)
+	s.EqualValues(0, describeResp.Info.MissedCatchupWindow)
+	s.EqualValues(0, describeResp.Info.OverlapSkipped)
+	s.EqualValues(0, len(describeResp.Info.RunningWorkflows))
+	s.EqualValues(2, len(describeResp.Info.RecentActions))
+	action0 := describeResp.Info.RecentActions[0]
+	s.WithinRange(*action0.ScheduleTime, createTime, time.Now())
+	s.True(action0.ScheduleTime.UnixNano()%int64(3*time.Second) == 0)
+	s.DurationNear(action0.ActualTime.Sub(*action0.ScheduleTime), 0, 500*time.Millisecond)
+
+	// list
+
+	s.Eventually(func() bool { // wait for visibility
+		listResp, err := s.engine.ListSchedules(NewContext(), &workflowservice.ListSchedulesRequest{
+			Namespace:       s.namespace,
+			MaximumPageSize: 5,
+		})
+		if err != nil || len(listResp.Schedules) != 1 || len(listResp.Schedules[0].Info.RecentActions) < 2 {
+			return false
+		}
+		s.NoError(err)
+		entry := listResp.Schedules[0]
+		s.Equal(sid, entry.ScheduleId)
+		s.Equal(schSAValue.Data, entry.SearchAttributes.IndexedFields[csa].Data)
+		s.Equal(schMemo.Data, entry.Memo.Fields["schedmemo1"].Data)
+		s.Equal(schedule.Spec, entry.Info.Spec)
+		s.Equal(wt, entry.Info.WorkflowType.Name)
+		s.False(entry.Info.Paused)
+		s.Equal(describeResp.Info.RecentActions, entry.Info.RecentActions) // 2 is below the limit where list entry might be cut off
+		return true
+	}, 3*time.Second, 200*time.Millisecond)
```

Comment:
I would suggest use a longer duration. 10s duration, with 1s interval, to reduce the flakiness.

Reply:
ok. do the other timeouts seem okay? I don't want it to be flaky but don't want to add that much to total integration test runtime either

Reply:
If you depends on visibility data, 3s could be flaky. 10s is much less likely to fail.